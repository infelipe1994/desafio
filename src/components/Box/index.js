import styled from 'styled-components'
import { border, color, flexbox, layout, space } from 'styled-system'

export const Box = styled.div`
  ${border};
  ${color};
  ${flexbox};
  ${layout};
  ${space};
`
