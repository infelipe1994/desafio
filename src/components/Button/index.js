import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Component = styled.button(
  ({ isFullWidth, isOutlined, theme }) => `
    appearance: none;
    align-items: center;
    background-color: ${isOutlined ? 'transparent' : theme.colors.green};
    border: 1px solid ${theme.colors.green};
    border-radius: 100rem;
    box-shadow: none;
    color: ${isOutlined ? theme.colors.green : theme.colors.smoke};
    cursor: pointer;
    display: inline-flex;
    font-size: ${theme.fontSizes.x1};
    font-weight: ${theme.fontWeights.x3};
    height: 2.5rem;
    justify-content: center;
    outline: none;
    padding: 0 ${theme.space.x4};
    position: relative;
    text-align: center;
    text-transform: uppercase;
    transition: all 0.250s;
    user-select: none;
    vertical-align: top;
    white-space: nowrap;

    &:hover, &:focus {
      background-color: ${theme.colors.green};
      border-color: ${theme.colors.green};
      color: ${theme.colors.smoke};
    }
    
    &:active {
      background-color: ${theme.colors.ocean};
      border-color: ${theme.colors.ocean};
      color: ${theme.colors.smoke};
    }

    &[disabled]{
      background-color: ${isOutlined ? 'transparent' : theme.colors.stone};
      border-color: ${theme.colors.stone};
      color: ${isOutlined ? theme.colors.stone : theme.colors.smoke};
      cursor: not-allowed;
    }

    ${isFullWidth ? 'display: flex; width: 100%' : ''};
  `
)

export const Button = p => <Component {...p} />

Button.propTypes = {
  isFullWidth: PropTypes.bool,
  isOutlined: PropTypes.bool
}
