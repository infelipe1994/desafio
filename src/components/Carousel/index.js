import React, { Children, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { layout } from 'styled-system'

import { ControlButtonGroup } from '@/src/components/Carousel/ControlButtonGroup'
import { SlideGroup } from '@/src/components/Carousel/Slide/Group'

const Component = styled.div`
  ${layout};

  position: relative;
`

export const Carousel = ({
  children,
  currentSlide,
  onChangeSlide,
  ...rest
}) => {
  const childrenCount = Children.count(children) - 1
  const [slide, setSlide] = useState(currentSlide)
  const handleNext = () => {
    const nextSlide = slide === childrenCount ? childrenCount : slide + 1

    onChangeSlide(nextSlide)
    setSlide(nextSlide)
  }
  const handlePrevious = () => {
    const nextSlide = slide === 0 ? 0 : slide - 1

    onChangeSlide(nextSlide)
    setSlide(nextSlide)
  }

  useEffect(
    () => document.getElementById(`${currentSlide}`).scrollIntoView(),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  )

  return (
    <Component width={rest.width}>
      <SlideGroup {...rest}>{children}</SlideGroup>
      <ControlButtonGroup
        handleNext={handleNext}
        handlePrevious={handlePrevious}
        slide={slide}
      />
    </Component>
  )
}

Carousel.defaultProps = {
  currentSlide: 0,
  onChangeSlide: () => {},
  width: '100%'
}

Carousel.propTypes = {
  children: PropTypes.node,
  currentSlide: PropTypes.number,
  onChangeSlide: PropTypes.func,
  width: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    ),
    PropTypes.number,
    PropTypes.string
  ])
}
