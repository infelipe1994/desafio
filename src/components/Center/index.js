import PropTypes from 'prop-types'
import styled from 'styled-components'
import { layout, space } from 'styled-system'

export const Center = styled.div`
  ${layout};
  ${space};

  box-sizing: content-box;
  margin-left: auto;
  margin-right: auto;

  ${({ alsoCenterChildren }) =>
    alsoCenterChildren
      ? 'align-items: center; display: flex; flex-direction: column'
      : ''};

  ${({ alsoCenterText }) => (alsoCenterText ? 'text-align: center' : '')};
`

Center.defaultProps = {
  maxWidth: '40em'
}

Center.propTypes = {
  alsoCenterChildren: PropTypes.bool,
  alsoCenterText: PropTypes.bool,
  maxWidth: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    ),
    PropTypes.number,
    PropTypes.string
  ])
}
