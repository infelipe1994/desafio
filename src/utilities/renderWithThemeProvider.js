import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'

import { theme } from '@/src/theme'

export const renderWithThemeProvider = Component =>
  render(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)
