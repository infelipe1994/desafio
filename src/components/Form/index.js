import React from 'react'
import PropTypes from 'prop-types'

import { Button } from '@/src/components/Button'
import { Stack } from '@/src/components/Stack'

export const Form = ({ children, disabled, onSubmit }) => (
  <Stack as="form" onSubmit={onSubmit} spaceBetween="x7">
    <Stack spaceBetween="x6">{children}</Stack>
    <Button disabled={disabled} isFullWidth>
      Avan√ßar
    </Button>
  </Stack>
)

Form.propTypes = {
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  onSubmit: PropTypes.func.isRequired
}
