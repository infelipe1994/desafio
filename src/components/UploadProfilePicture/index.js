import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { Button } from '@/src/components/Button'
import { ProfilePicture } from '@/src/components/UploadProfilePicture/ProfilePicture'
import { Stack } from '@/src/components/Stack'

export const UploadProfilePicture = ({ onUploadPicture, pictureSrc }) => {
  const [src, setSrc] = useState(pictureSrc)

  return (
    <Stack direction="row" spaceBetween="x3">
      <ProfilePicture setSrc={setSrc} src={src} />
      <Button
        disabled={!src}
        isOutlined
        onClick={() => onUploadPicture(src)}
        type="button"
      >
        Enviar foto
      </Button>
    </Stack>
  )
}

UploadProfilePicture.defaultProps = {
  pictureSrc: ''
}

UploadProfilePicture.propTypes = {
  onUploadPicture: PropTypes.func.isRequired,
  pictureSrc: PropTypes.string
}
