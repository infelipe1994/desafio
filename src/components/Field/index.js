import React, { useRef } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { theme } from '@/src/theme'
import { getColor } from '@/src/utilities/getColor'
import { isDateType } from '@/src/utilities/isDateType'
import { placeholder } from '@/src/utilities/placeholder'
import { uuid } from '@/src/utilities/uuid'

const Input = styled.input(
  ({ color }) => `
    appearance: none;
    background-color: transparent;
    border: 0;
    color: ${color};
    font-size: ${theme.fontSizes.x3};
    height: ${theme.lineHeights.x3};
    line-height: ${theme.lineHeights.x3};
    outline: 0;
    transition: all 0.250s;
    width: 100%;

    &[disabled] {
      cursor: not-allowed;
    }

    ${placeholder(color)};
  `
)

const Container = styled.div(
  ({ borderColor, error }) => `
    border-bottom: 1px ${error ? 'dashed' : 'solid'} ${borderColor};
    padding-bottom: ${theme.space.x1};
    position: relative;
    transition: all 0.250s;
    width: 100%;

    & input:not(:placeholder-shown) + label {
      opacity: 1;
      transform: translateY(-${theme.lineHeights.x3});
    }
  `
)

const Label = styled.label(
  ({ color, disabled }) => `
    bottom: ${theme.space.x1};
    color: ${color};
    font-size: ${theme.fontSizes.x3};
    left: 0;
    line-height: ${theme.lineHeights.x3};
    opacity: 0;
    position: absolute;
    transition: all 0.250s;

    ${disabled ? 'cursor: not-allowed' : ''};
  `
)

export const Field = ({ disabled, error, inputRef, label, type, ...rest }) => {
  const colors = getColor(disabled, error, theme)
  const id = useRef(rest.id || uuid())
  const isDate = isDateType(type)
  const stateProps = { ...colors, disabled, error }

  return (
    <Container {...stateProps}>
      <Input
        {...rest}
        {...stateProps}
        id={id.current}
        placeholder={label}
        ref={inputRef}
        type={isDate ? 'text' : type}
      />
      <Label {...stateProps} htmlFor={id.current}>
        {label}
      </Label>
    </Container>
  )
}

Field.defaultProps = {
  type: 'text'
}

Field.propTypes = {
  disabled: PropTypes.bool,
  error: PropTypes.bool,
  inputRef: PropTypes.func,
  label: PropTypes.string.isRequired,
  type: PropTypes.string
}
