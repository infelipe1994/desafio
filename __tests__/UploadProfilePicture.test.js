import React from 'react'
import { cleanup } from '@testing-library/react'

import { UploadProfilePicture } from 'src/components/UploadProfilePicture'
import { renderWithThemeProvider } from 'src/utilities/renderWithThemeProvider'

afterEach(cleanup)

describe('UploadProfilePicture', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithThemeProvider(
      <UploadProfilePicture onUploadPicture={jest.fn()} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
