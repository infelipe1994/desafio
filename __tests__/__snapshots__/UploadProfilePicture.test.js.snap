// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProfilePicture should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Stack__Component-i9fw1o-0 hReGPj"
  >
    <div
      class="Box-vhihpg-0 jGEIOo"
      display="flex"
      height="4.500rem"
      style="cursor: pointer;"
      width="4.500rem"
    >
      <label
        for="48d5a389-9233-41e7-b267-8fa6868e6332"
        style="cursor: pointer; line-height: 0;"
      >
        <svg
          aria-hidden="true"
          aria-label="Selecione uma foto de perfil."
          fill="none"
          focusable="false"
          role="img"
          style="height: 3.500rem; width: 3.500rem;"
          viewBox="0 0 55 55"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M53 53V47C53 40.3726 47.2916 35 40.25 35H14.75C7.70837 35 2 40.3726 2 47V53"
            stroke="#c2d6d7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
          />
          <path
            clip-rule="evenodd"
            d="M27.5 29C34.9558 29 41 22.9558 41 15.5C41 8.04416 34.9558 2 27.5 2C20.0442 2 14 8.04416 14 15.5C14 22.9558 20.0442 29 27.5 29Z"
            fill-rule="evenodd"
            stroke="#c2d6d7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
          />
        </svg>
        <input
          hidden=""
          id="48d5a389-9233-41e7-b267-8fa6868e6332"
          type="file"
        />
      </label>
    </div>
    <button
      class="Button__Component-sc-1mv13uc-0 IMdNu"
      disabled=""
      type="button"
    >
      Enviar foto
    </button>
  </div>
</DocumentFragment>
`;
